torch/csrc/autograd/VariableTypeManual.cpp

```diff
@@ -194,6 +195,52 @@ void retain_grad(Tensor & self) {
   impl::get_autograd_meta(self)->retains_grad_ = true;
 }
 
+// Taken from codegened version
+Tensor fw_primal(const Tensor & self, int64_t level) {
+  auto& self_ = unpack(self, "self", 0);
+  std::shared_ptr<FwPrimalBackward> grad_fn;
+  if (compute_requires_grad( self )) {
+    grad_fn = std::shared_ptr<FwPrimalBackward>(new FwPrimalBackward(), deleteNode);
```

Comment:
why not make_shared?

Reply:
This comes from the codegen as well. Not sure why it does it this way. To have the custom deleter maybe?